dali/pipeline/operator/builtin/external_source.h

```diff
@@ -216,7 +218,13 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
         previous_dtype_(DALIDataType::DALI_NO_TYPE),
         ndim_(-1),
         layout_(),
-        sync_worker_(device_id_, false, "ExternalSource syncworker") {
+        sync_worker_(device_id_, false, "ExternalSource syncworker"),
+        internal_copy_stream_(std::is_same<Backend, CPUBackend>::value ?
+                                  CUDAStreamLease{} :
```

Comment:
You have to store the lease object, otherwise it will be given back to the pool and the stream will be reused by some unrelated party, possibly skewing synchronization.